internal/applicationsnapshot/input.go

```diff
@@ -176,3 +197,76 @@ func readSnapshotSource(input []byte) (app.SnapshotSpec, error) {
 	log.Debugf("Read application snapshot from file %s", input)
 	return file, nil
 }
+
+func (*remoteClient) Get(ref name.Reference) (*remote.Descriptor, error) {
+	return remote.Get(ref)
+}
+
+func (*remoteClient) Index(ref name.Reference) (v1.ImageIndex, error) {
+	return remote.Index(ref)
+}
+
+func clientFromContext(ctx context.Context) (client, bool) {
+	cli, ok := ctx.Value(RemoteClientKey{}).(client)
+	return cli, ok
+}
```

Comment:
Perhaps this; it would make the caller unaware of the client selection

```suggestion
var defaultClient = &remoteClient{}

func determineClient(ctx context.Context) client {
	if cli, ok := ctx.Value(RemoteClientKey{}).(client); ok {
		return cli
	}

	return defaultClient
}
```